File: knowledge/scaffolds/libraries/java-maven/src/main/java/com/example/mylib/MyLib.java
package com.example.mylib;

/**
 * MyLib provides reusable utility methods.
 */
public final class MyLib {

    private MyLib() {
        // Utility class — no instantiation
    }

    /**
     * Add two integers together.
     *
     * @param a first number
     * @param b second number
     * @return the sum of a and b
     */
    public static int add(int a, int b) {
        return a + b;
    }

    /**
     * Generate a greeting string with the default greeting "Hello".
     *
     * @param name the name to greet
     * @return the greeting string
     * @throws IllegalArgumentException if name is null or empty
     */
    public static String greet(String name) {
        return greet(name, "Hello");
    }

    /**
     * Generate a greeting string with a custom greeting.
     *
     * @param name     the name to greet
     * @param greeting the greeting prefix
     * @return the greeting string
     * @throws IllegalArgumentException if name is null or empty
     */
    public static String greet(String name, String greeting) {
        if (name == null || name.isEmpty()) {
            throw new IllegalArgumentException("name must not be empty");
        }
        return greeting + ", " + name + "!";
    }

    /**
     * Clamp a value to lie within [min, max].
     *
     * @param value the value to clamp
     * @param min   the minimum bound
     * @param max   the maximum bound
     * @return the clamped value
     */
    public static int clamp(int value, int min, int max) {
        return Math.max(min, Math.min(max, value));
    }
}

File: knowledge/scaffolds/libraries/ruby-gem/lib/mylib.rb
# frozen_string_literal: true

require_relative "mylib/version"

# MyLib provides reusable utility functions.
module MyLib
  class Error < StandardError; end

  # Add two numbers together.
  #
  # @param a [Numeric] first number
  # @param b [Numeric] second number
  # @return [Numeric] the sum
  def self.add(a, b)
    a + b
  end

  # Generate a greeting string.
  #
  # @param name [String] the name to greet
  # @param greeting [String] the greeting prefix (default: "Hello")
  # @return [String] the greeting
  # @raise [Error] if name is empty
  def self.greet(name, greeting: "Hello")
    raise Error, "name must not be empty" if name.nil? || name.empty?

    "#{greeting}, #{name}!"
  end
end
